lib/charms/oathkeeper/v0/oathkeeper_info.py

```diff
@@ -0,0 +1,194 @@
+    def some_event_function():
+        # fetch the relation info
+        if self.oathkeeper_info_relation.is_ready():
+            oathkeeper_data = self.oathkeeper_info_relation.get_oathkeeper_info()
+```
+"""
+
+import logging
+from typing import Dict, Optional
+
+from ops.charm import CharmBase, RelationChangedEvent
+from ops.framework import EventBase, EventSource, Object, ObjectEvents
+
+# The unique Charmhub library identifier, never change it
+LIBID = "c801a227f45b46099d7f87cff2dc6e39"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+RELATION_NAME = "oathkeeper-info"
+INTERFACE_NAME = "oathkeeper_info"
+logger = logging.getLogger(__name__)
+
+
+class OathkeeperInfoRelationChangedEvent(EventBase):
+    """Event to notify the charm that the relation data has changed."""
+
+
+class OathkeeperInfoProviderEvents(ObjectEvents):
+    """Event descriptor for events raised by `OathkeeperInfoProvider`."""
+
+    data_changed = EventSource(OathkeeperInfoRelationChangedEvent)
+
+
+class OathkeeperInfoProvider(Object):
+    """Provider side of the oathkeeper-info relation."""
+
+    on = OathkeeperInfoProviderEvents()
+
+    def __init__(self, charm: CharmBase, relation_name: str = RELATION_NAME):
+        super().__init__(charm, relation_name)
+
+        self._charm = charm
+        self._relation_name = relation_name
+
+        events = self._charm.on[relation_name]
+        self.framework.observe(events.relation_changed, self._on_info_provider_relation_changed)
+
+    def _on_info_provider_relation_changed(self, event: RelationChangedEvent) -> None:
+        self.on.data_changed.emit()
+
+    def send_info_relation_data(
+        self,
+        public_endpoint: str,
+        rules_configmap_name: str,
+        configmaps_namespace: str,
+    ) -> None:
+        """Updates relation with endpoints and configmaps info."""
+        if not self._charm.unit.is_leader():
+            return
+
+        relations = self.model.relations[self._relation_name]
+        info_databag = {
+            "public_endpoint": public_endpoint,
+            "rules_configmap_name": rules_configmap_name,
+            "configmaps_namespace": configmaps_namespace,
+        }
+
+        for relation in relations:
+            relation.data[self._charm.app].update(info_databag)
+
+    def get_requirer_info(self) -> Optional[str]:
+        """Get the requirer info."""
+        info = self.model.relations[self._relation_name]
+        if len(info) == 0:
+            return None
+
+        if not (app := info[0].app):
+            return None
+
+        data = info[0].data[app]
+
+        if not data:
+            logger.info("No relation data available.")
+            return None
+
+        return data["access_rules_file"]
```

Comment:
question: this question is actually more for me than for the review. Do we know for sure that this field will be present, or is there even the slightest chance that it could be missing? Given my limited charm knowledge, I would approach this with either a `data.get('access_rules_file')` (so it could return `None` in case the key is not present) or by preluding this with a check to see if the value is actually there.